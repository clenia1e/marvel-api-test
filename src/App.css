* {
  box-sizing: border-box;
}

.root {
  max-width: 1440px;
  min-height: 100%;
  padding: 0 60px;
  margin: 0 auto;
  width: 100%;
  position: relative;
}
