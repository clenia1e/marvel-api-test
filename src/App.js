import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import "./App.css";
import Character from "./Character";
import Home from "./Home";

function App() {
  return (
    <>
      <div className="root">
        <Router>
          <Switch>
            <Route path="/character">
              <Character />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </Router>
      </div>
    </>
  );
}

export default App;
