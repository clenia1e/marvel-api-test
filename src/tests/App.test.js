import { render } from "@testing-library/react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import App from "../App";

const server = setupServer(
  rest.get(
    "https://gateway.marvel.com/v1/public/characters",
    (req, res, ctx) => {
      return res(
        ctx.status(200),
        ctx.json({
          data: {
            offset: 0,
            limit: 20,
            total: 1,
            count: 1,
            results: [
              {
                id: 1011334,
                name: "test man",
                description: "",
                modified: "2014-04-29T14:18:17-0400",
                thumbnail: {
                  path: "http://testimage.com/12378912",
                  extension: "jpg",
                },
              },
            ],
          },
        })
      );
    }
  )
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("renders home page", async () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
