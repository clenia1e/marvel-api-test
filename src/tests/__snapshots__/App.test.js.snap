// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="header-root container-header"
    >
      <div
        class="container-logo"
      >
        <img
          alt="ícone da Marvel"
          class="img-logo"
          src="logo.svg"
        />
      </div>
      <h2
        class="main-text"
      >
        EXPLORE O UNIVERSO
      </h2>
      <p
        class="secundary-text"
      >
        Mergulhe no domínio deslumbrante de todos os personagens clássicos que você ama - e aqueles que você descobrirá em breve!
      </p>
    </div>
    <div
      class="container-search"
    >
      <div
        class="container-wrapper"
      >
        <div
          class="container-icon"
        >
          <img
            alt="ícone de navegacao"
            src="/ic_busca_menor.svg"
          />
        </div>
        <input
          class="container-searchbar--red"
          data-testid="search-input"
          placeholder="Procure por heróis"
          value=""
        />
      </div>
    </div>
    <div
      class="container-info"
    >
      <div
        class="info-length"
      >
        Encontrados 
        0
         heróis
      </div>
      <div
        class="info-buttons"
      >
        <div
          class="order-button"
        >
          <img
            alt="ícone de navegacao"
            class="img-hero"
            src="ic_heroi.svg"
          />
          Ordenar por nome - A/Z
        </div>
        <img
          alt="ícone de seleção"
          class="img-toggle"
          data-testid="toggle-asc"
          role="presentation"
          src="toggle_on.svg"
        />
        <div
          class="favorite-button"
        >
          <button
            class="button-favorite"
            data-testid="toggle-favorite"
          >
            <img
              alt="ícone de navegacao"
              class="img-favorite"
              src="favorito_01.svg"
            />
            Somente favoritos
          </button>
        </div>
      </div>
    </div>
    <div
      class="container-loader"
    >
       
      <div
        class="loader"
      />
    </div>
    <div
      class="pagination"
    >
      <button
        class="prev disabled"
      >
        prev
      </button>
      <button
        class="next "
        data-testid="pagination-next"
      >
        next
      </button>
    </div>
    <div
      class="footer"
    />
  </div>
</div>
`;
